app/src/features/apiClient/contexts/apiClient.tsx

```diff
@@ -0,0 +1,104 @@
+import { getUserAuthDetails } from "store/selectors";
+import { RQAPI } from "../types";
+import { getApiRecords } from "backend/apiClient";
+import Logger from "lib/logger";
+
+interface ApiClientContextInterface {
+  apiClientRecords: RQAPI.Record[];
+  isLoadingApiClientRecords: boolean;
+  onNewRecord: (apiClientRecord: RQAPI.Record) => void;
+  onRemoveRecord: (apiClientRecord: RQAPI.Record) => void;
+  onUpdateRecord: (apiClientRecord: RQAPI.Record) => void;
+  onSaveRecord: (apiClientRecord: RQAPI.Record) => void;
+}
+
+const ApiClientContext = createContext<ApiClientContextInterface>({
+  apiClientRecords: [],
+  isLoadingApiClientRecords: false,
+  onNewRecord: (apiClientRecord: RQAPI.Record) => {},
+  onRemoveRecord: (apiClientRecord: RQAPI.Record) => {},
+  onUpdateRecord: (apiClientRecord: RQAPI.Record) => {},
+  onSaveRecord: (apiClientRecord: RQAPI.Record) => {},
+});
+
+interface ApiClientProviderProps {
+  children: React.ReactElement;
+}
+
+export const ApiClientProvider: React.FC<ApiClientProviderProps> = ({ children }) => {
+  const user = useSelector(getUserAuthDetails);
+  const uid = user?.details?.profile?.uid;
+  const workspace = useSelector(getCurrentlyActiveWorkspace);
+  const teamId = workspace?.id;
+
+  const [isLoadingApiClientRecords, setIsLoadingApiClientRecords] = useState(false);
+  const [apiClientRecords, setApiClientRecords] = useState<RQAPI.Record[]>([]);
+
+  useEffect(() => {
+    if (!uid) {
+      return;
+    }
+
+    setIsLoadingApiClientRecords(true);
+    getApiRecords(uid, teamId)
+      .then((result) => {
+        if (result.success) {
+          setApiClientRecords(result.data);
+        }
+      })
+      .catch((error) => {
+        setApiClientRecords([]);
+        Logger.error("Error loading api records!", error);
+      })
+      .finally(() => {
+        setIsLoadingApiClientRecords(false);
+      });
+  }, [uid, teamId]);
+
+  const onNewRecord = useCallback((apiClientRecord: RQAPI.Record) => {
+    setApiClientRecords((prev) => {
+      return [...prev, { ...apiClientRecord }];
+    });
+  }, []);
+
+  const onRemoveRecord = useCallback((apiClientRecord: RQAPI.Record) => {
+    setApiClientRecords((prev) => {
+      return prev.filter((record) => record.id !== apiClientRecord.id);
+    });
+  }, []);
+
+  const onUpdateRecord = useCallback((apiClientRecord: RQAPI.Record) => {
+    setApiClientRecords((prev) => {
+      return prev.map((record) => (record.id === apiClientRecord.id ? { ...record, ...apiClientRecord } : record));
+    });
+  }, []);
+
+  const onSaveRecord = useCallback(
+    (apiClientRecord: RQAPI.Record) => {
+      const isRecordExist = apiClientRecords.find((record) => record.id === apiClientRecord.id);
+
+      if (isRecordExist) {
+        onUpdateRecord(apiClientRecord);
+      } else {
+        onNewRecord(apiClientRecord);
+      }
+    },
+    [apiClientRecords, onUpdateRecord, onNewRecord]
+  );
```

Comment:
Can we name these functions? These functions are simply state updates
What about: `addApiRecord `, `updateApiRecord `, `removeApiRecord `